### ISSUE TITLE
setTimeout in test cases

### ISSUE THREAD

**Issue by kitsonk** [dojo/widgets#5]
The only way currently we are able to easily test certain async processes in widgets is using setTimeout.  Obviously this is fragile and should be fixed.

**Comment by vansimke** [comment]
While I wasn't able to figure out a way around the use of setTimeout, I have submitted a PR for a [POC](https://github.com/dojo/widgets/pull/99) to add a wrapper to setTimeout to stabilize its behavior without sacrificing test suite execution time.

@kitsonk can you take a look and give feedback? If we're okay with this direction, I can update the rest of the test suite to follow this pattern.

**Comment by kitsonk** [comment]
@matt-gadd and @ant it would be good to get your 2p as well

**Comment by vansimke** [comment]
I thought I had everything done in this task, but I realized that there was a 'delay()' helper method that was wrapping a setTimeout call with a promise. As I worked my through those, I ran into major problems with the tests in createStatefulListenersMixin. I can't find anything to hook into to determine if the async operation is done that is any better than a straight setTimeout. However, I am thinking that dojo/compose#54 might offer a solution to this.
